src/devices/machine/bitmap_printer.h

```diff
@@ -47,43 +47,47 @@ class bitmap_printer_device : public device_t
 
 private:
 	required_device<screen_device> m_screen;
-	required_device<session_time_device> m_session_time;
+//  required_device<session_time_device> m_session_time;
 public:
 	required_device<stepper_device> m_pf_stepper;
 	required_device<stepper_device> m_cr_stepper;
 
-	int m_cr_direction = 1; // direction of carriage
-	int m_pf_stepper_ratio0 = 1;
-	int m_pf_stepper_ratio1 = 1;
-	int m_cr_stepper_ratio0 = 1;
-	int m_cr_stepper_ratio1 = 1;
-	int m_xpos = 0;
-	int m_ypos = 0;
+	int m_cr_direction; // direction of carriage
+	int m_pf_stepper_ratio0;
+	int m_pf_stepper_ratio1;
+	int m_cr_stepper_ratio0;
+	int m_cr_stepper_ratio1;
+	int m_xpos;
+	int m_ypos;
 
 	bitmap_rgb32  m_page_bitmap;  // page bitmap
 private:
-	const int PAPER_SCREEN_HEIGHT = 384; // match the height of the apple II driver
-	const int m_distfrombottom = 50;  // print position from bottom of screen
-
-	int m_printhead_color       = 0xEEE8AA;
-	int m_printhead_bordercolor = 0xBDB76B;
-	int m_printhead_bordersize = 2;
-	int m_printhead_xsize = 10;
-	int m_printhead_ysize = 20;
-	int m_pagedirty = 0;
-	int m_paperwidth;
-	int m_paperheight;
+	static constexpr int PAPER_SCREEN_HEIGHT = 384; // match the height of the apple II driver
+	static constexpr int m_distfrombottom = 50;  // print position from bottom of screen
+
+	int m_printhead_color;
+	int m_printhead_bordercolor;
+	int m_printhead_bordersize;
+	int m_printhead_xsize;
+	int m_printhead_ysize;
+	int m_page_dirty;
+	int m_paper_width;
+	int m_paper_height;
 	int m_hdpi;
 	int m_vdpi;
-	int clear_pos = 0;
-	int newpageflag = 0;  // used to keep printhead at the top of page until actual printing
+	int m_clear_pos;
+	int m_newpage_flag;  // used to keep printhead at the top of page until actual printing
+	static constexpr int MAX_LEDS = 5;
+	int m_led_state[MAX_LEDS];
+	int m_num_leds;
 
 public:
+	void set_led_state(int led, int value) { m_led_state[led] = value; m_num_leds = std::max(m_num_leds, led); }
```

Comment:
If the LEDs have consistent meanings, define an public enum inside the class to give them meaningful names like bitmap_printer_device::LED_READY and stuff like that.